### ISSUE TITLE
How does file contents are refreshed and does the driver support zero down time updates?

### ISSUE THREAD

**Issue by ozancaglayan** [awslabs/mountpoint-s3-csi-driver#274]
Hello,

- If files are updated in the relevant S3 buckets, how quick the changes are reflected to the nodes which had the bucket mounted through static provisioning?
- When updating the driver version in the cluster, do pods risk of losing access to the mount points or is it zero down time?
Thanks

/triage support

**Comment by monthonk** [comment]
Hey, to answer your questions
1. It depends on your mount configurations and file system operations. If you do not enable any caching then the changes should be reflected immediately in most cases. Please see more details in [Mountpoint's consistency and concurrency](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#consistency-and-concurrency).
2. The running pods will keep using existing version of Mountpoint so won’t be losing the access to the file system. Only new pods will start using the new version of CSI Driver and Mountpoint.

**Comment by monthonk** [comment]
I'm closing this issue. Please re-open if needed.

**Comment by ozancaglayan** [comment]
Thanks a lot for the info!

**Comment by ozancaglayan** [comment]
Hi, sorry to ping from here but I have another question which annoys me

When I update the IAM role/policy to give access to a new S3 bucket, I can't make the thing work. CSI driver falls back to worker node role. I uninstalled the driver, and re-installed it, but still weirdly, its not giving me access to the new bucket. I saw that there's an expiration_seconds for CSIDriver. Is this relevant? Should the changes to the IAM policy be reflected immediately or only after 3600 seconds? Is there a way to invalidate the tokens so that they're fetched again?

thanks

**Comment by unexge** [comment]
Hey @ozancaglayan, what authentication mechanism are you using? The CSI driver might be falling back to the instance role if the other methods (i.e., [long-term AWS credentials](https://github.com/awslabs/mountpoint-s3-csi-driver/blob/main/docs/CONFIGURATION.md#driver-level-credentials-with-k8s-secrets) and [Driver-level IRSA](https://github.com/awslabs/mountpoint-s3-csi-driver/blob/main/docs/CONFIGURATION.md#driver-level-credentials-with-irsa)) are failing. IAM policy changes should be reflected instantly (maybe with some small delay on IAM service).

**Comment by ozancaglayan** [comment]
To be honest, no idea =) I'm just installing the driver with:

```bash
  helm upgrade --install aws-mountpoint-s3-csi-driver \
    --namespace kube-system \
    --set node.serviceAccount.annotations."eks\.amazonaws\.com/role-arn"="${{ env.role_arn }}" \
    --set node.tolerateAllTaints='true' \
    aws-mountpoint-s3-csi-driver/aws-mountpoint-s3-csi-driver \
    --version ${{ inputs.version }}
```

I'm not injecting any credentials. Upon inspecting the daemonset in a node, I can see that:
- `AWS_WEB_IDENTITY_TOKEN_FILE` is defined and points to `/var/run/secrets/...`

Maybe I'm missing a step during installation but everything is actually working but I dont know when they kick in. This time, after I added a new bucket to the IAM, I couldn't even make it work after driver uninstall. Either I'm not seeing a very obvious error or typo or I'm doing sth wrong in terms of authentication.

It was also falling back to the instance role but then I was simply putting in a wrong `role-arn` above. Once I've fixed it, it started to work.

Thanks a lot, appreciate it.
